File: algorithms/Strings/Easy/RobotReturnToOrigin/VSDRobotReturnToOrigin.java


public class Main{

    static boolean returnOrigin(String moves){
        int x=0,y=0;
        for(int i=0;i<moves.length();i++){
            switch(moves.charAt(i)){
                case 'U':y++;break;
                case 'D':y--;break;
                case 'R':x++;break;
                default:x--;
            }
        }
        if(x==0 && y==0) return true;
        return false;
    }
    public static void main(String[] args) {
        String moves="UD";
        System.out.println(returnOrigin(moves));
    }
}

File: algorithms/Strings/Easy/RobotReturnToOrigin/VSDRobotReturnToOrigin.py



def judgeCircle(moves: str) -> bool:

    x_coord,y_coord=0,0
    for move in moves:
        if move=="U":
            y_coord+=1
        elif move=="D":
            y_coord-=1
        elif move=="L":
            x_coord-=1
        else:
            x_coord+=1
    return x_coord==0 and y_coord==0


if "__name__"=="__main__":
    moves="UD"
    print(judgeCircle(moves))